Packs/AsanaConnect/Integrations/AsanaConnect/AsanaConnect.py

```diff
@@ -0,0 +1,153 @@
+
+    :type id: ``str``
+    :param id: project id in Asana
+
+    :type name: ``str``
+    :param name: name for the task to be created
+
+    :type token: ``str``
+    :param token: PAT of the account used in this integration
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+    project = {}
+    try:
+        client = get_asana_client(token)
+        project = client.projects.find_by_id(pid)
+        return CommandResults(
+            outputs_prefix='asana',
+            outputs_key_field='project',
+            outputs=project
+        )
+    except Exception as e:
+        demisto.results("Request failed")
+
+
+def create_task_in_project(pid, name, token):
+    """Creates a task in the specific Asana project'
+
+    :type id: ``str``
+    :param id: project id in Asana
+
+    :type name: ``str``
+    :param name: name for the task to be created
+
+    :type token: ``str``
+    :param token: PAT of the account used in this integration
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+    try:
+        client = get_asana_client(token)
+        client.tasks.create({'projects': pid, 'name': name})
+        demisto.results(f'Task {name} successfully added to project')
+    except Exception as e:
+        demisto.results('Task creation failed')
+
+
+def test_module(token):
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Shows the http error 'No Authorization' if it fails.
+
+    :type token: ``str``
+    :param token: PAT of the Asana account used in this integration
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+    get_asana_client(token)
+
+    return 'ok'
+
+
+def main():
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    params = demisto.params()
+    access_token = params.get('accesstoken')
+    try:
+        if demisto.command() == 'test-module':
+            demisto.results(test_module(access_token))
+
+        elif demisto.command() == 'asana-get-project':
+            pid = demisto.args()['project_id']
+            res = get_project(pid, access_token)
+            return_results(res)
+
+        elif demisto.command() == 'asana-create-task':
+            pid = demisto.args()['project_id']
+            name = demisto.args()['name']
+            create_task_in_project(pid, name, access_token)
+
+        elif demisto.command() == 'asana-get-all-projects':
+            res = get_all_projects(access_token)
+            return_results(res)
+    except Exception as err:
+        if isinstance(err, NotImplementedError) and COMMAND_NOT_IMPELEMENTED_MSG in str(err):
+            raise
+        return_error(str(err))
+
+    finally:
+        LOG.print_log()
```

Comment:
```suggestion
        LOG.print_log()
```

What is this for? not sure why you need it

Reply:
a logger for my own dev environment. I have removed it.